src/telemetry-tracing.ts

```diff
@@ -423,15 +522,28 @@ export class PubsubSpans {
 
   static createReceiveLeaseSpan(
     message: MessageWithAttributes,
-    deadline: Duration,
-    isInitial: boolean
+    subName: string,
+    type: 'modack' | 'ack' | 'nack',
+    deadline?: Duration,
+    isInitial?: boolean
   ): Span | undefined {
-    const span = PubsubSpans.createChildSpan('modify ack deadline', message);
-    span?.setAttribute(
-      'messaging.gcp_pubsub.modack_deadline_seconds',
-      deadline.totalOf('second')
+    const subInfo = getSubscriptionInfo(subName);
+    const attributes = this.createAttributes(subInfo, message);
+    const span = PubsubSpans.createChildSpan(
```

Comment:
These RPC spans should be in their own trace as a root span rather than a child span. Maybe I'm misunderstanding what this function does?

Reply:
Yeah, the function name is a bit confusing I guess.

```typescript
    const span = PubsubSpans.createChildSpan(
      `${subInfo.subId} ${type}`,
      message,
      undefined,
      attributes
    );
```

The `undefined` there specifies a parent span, or in this case, no parent.

Reply:
Actually, looking at `createChildSpan`, this isn't correct. I'll fix it.